### PR TITLE
[SYCL][Joint Matrix] Refactor/fix several JM tests. Remove xfail.

### PR DESCRIPTION
**Pull request by YuriPlyakhin**
Remove xfail for get coord tests and 2 bf16-related tests for Joint Matrix after enabling this functionality on PVC with sub group size =32. Fix tests to work correctly with different sub groups. Remove unnecessary code and comments.

### CODE REVIEW

File: sycl/test-e2e/Matrix/get_coord_bf16_matC_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  matrix_rand(MATRIX_M, MATRIX_N, (float *)&C, (float)100);
+  matrix_sum_rows(MC, sum_rows);
+
+  bool res = true;
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      sum_rows_ref[i] += C[i][j];
+    }
+    if (fabs(sum_rows_ref[i] - sum_rows[i]) > BF16_EPSILON)
```
Comment (dkhaldi) [inline sycl/test-e2e/Matrix/get_coord_bf16_matC_impl.hpp:96]: Remove BF16_EPSILON --> zero as these are float

File: sycl/test-e2e/Matrix/get_coord_bf16_matC_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
+//==----------- get_coord_bf16_matC_impl.hpp  - DPC++ joint_matrix---------==//
```
Comment (dkhaldi) [inline sycl/test-e2e/Matrix/get_coord_bf16_matC_impl.hpp:1]: This test should be called get_coord_float_matC then, right?

File: sycl/test-e2e/Matrix/get_coord_bf16_matC_impl.hpp (hunk 1 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/get_coord_bf16_matC_impl.hpp]: float still needs some epsilon. Replaced with new FLOAT_EPSILON

File: sycl/test-e2e/Matrix/get_coord_bf16_matC_impl.hpp (hunk 2 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/get_coord_bf16_matC_impl.hpp]: Yep, good catch :)

File: sycl/test-e2e/Matrix/SG32/get_coord_bf16_matC.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-//==----------- get_coord_bf16_gemm.cpp  - DPC++ joint_matrix---------==//
+//==----------- get_coord_bf16_matC.cpp  - DPC++ joint_matrix---------==//
 //
```
[the remark this answers is not in the document]
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/SG32/get_coord_bf16_matC.cpp]: fixed

**Comment by YuriPlyakhin** [comment]
@intel/llvm-gatekeepers , please, merge. Fails are unrelated to this patch.

There is open https://github.com/intel/llvm/issues/10460
AMD/HIP: Failed Tests (2):
  SYCL :: AtomicRef/store.cpp
  SYCL :: GroupAlgorithm/exclusive_scan_over_group.cpp
   ```Memory access fault by GPU node-1 (Agent handle: 0x223ffb0) on address 0x7f42b0400000. Reason: Page not present or supervisor privilege.```

**Comment by asudarsa** [comment]
LGTM. Thanks.

**Pull request merged**